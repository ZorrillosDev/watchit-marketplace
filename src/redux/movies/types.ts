export interface MoviesArgs {
  limit?: number
}
