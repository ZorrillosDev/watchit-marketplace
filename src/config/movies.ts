export const FAKE_MOVIES = [
  {
    creator: {
      address: '0x0010',
      movies: 4
    },
    owner: {
      address: '0x0022',
      movies: 9
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/11/help-50817-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0xEe99CeFF640d37Edd9cac8c7cfF4Ed4cD609f435',
      movies: 10
    },
    owner: {
      address: '0x0054',
      movies: 7
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/08/safer-at-home-46847-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0x005',
      movies: 5
    },
    owner: {
      address: '0x006',
      movies: 3
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/11/cagefighter-worlds-collide-50877-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  },
  {
    creator: {
      address: '0x007',
      movies: 2
    },
    owner: {
      address: '0x008',
      movies: 12
    },
    image: 'https://cuevana3.io/wp-content/uploads/2021/10/este-nino-es-un-trasto-50575-poster-200x300.jpg',
    name: 'Help',
    description: 'Tras los sucesos de “Vengadores: Infinity War”, los superhéroes que sobrevivieron a Thanos se reunen para poner en práctica un plan definitivo que podría acabar con el villano definitivamente. No saben si funcionará, pero es su única oportunidad de intentarlo. Cuarta entrega de la saga “Vengadores”',
    price: 1.8,
    rating: 20
  }
]
