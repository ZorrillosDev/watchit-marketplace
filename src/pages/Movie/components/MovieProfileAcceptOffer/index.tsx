// ===========================|| ACCEPT OFFER EXPORTS ||=========================== //

export {
  MovieProfileAcceptOffers as default
} from '@pages/Movie/components/MovieProfileAcceptOffer/MovieProfileAcceptOfferContainer'
