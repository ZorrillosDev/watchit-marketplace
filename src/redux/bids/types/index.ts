export * from './bids';
export * from './bids.actions';
export * from './bids.state';
