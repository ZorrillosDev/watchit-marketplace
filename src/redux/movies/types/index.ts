export * from './movies'
export * from './movies.actions'
export * from './movies.state'