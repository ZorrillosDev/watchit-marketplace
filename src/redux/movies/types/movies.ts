export const GenresList = [
  'All',
  'Action',
  'Adventure',
  'Animation',
  'Biography',
  'Comedy',
  'Crime',
  'Documentary',
  'Drama',
  'Family',
  'Fantasy',
  'Film-Noir',
  'History',
  'Horror',
  'Music',
  'Musical',
  'Mystery',
  'Romance',
  'Sci-Fi',
  'Sport',
  'Thriller',
  'War',
  'Western',
  'News',
  'Reality-TV',
  'Talk-Show',
  'Game-Show'
] as const

export type Genres = typeof GenresList[number]

export interface Images {
  small: string
  medium: string
  large: string
}

export interface MovieBid {
  id: string
  account: string
  created_at: string
  bid: number
}

export interface Movie {
  creator: string
  description: string
  path: string
  title: string
  date_uploaded_unix: number
  price: number
  group_name: string
  imdb_code: string
  language: string
  mpa_rating: string
  rating: number
  runtime: number
  synopsis: string
  trailer_code: string
  year: number
  genres: Genres
  posters: Images
}
