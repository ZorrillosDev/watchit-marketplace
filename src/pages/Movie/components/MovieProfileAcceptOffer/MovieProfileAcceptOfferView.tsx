// REACT IMPORTS
import React, { FC, useState } from 'react'

// MUI IMPORTS
import { Box, BoxProps, Button, Grid, styled, Alert, Theme, Typography } from '@mui/material'
import { LoadingButton } from '@mui/lab'
import { SxProps } from '@mui/system'

// PROJECT IMPORTS
import Modal from '@components/Modal'
import { Translation } from '@src/i18n'
import {
  MovieProfileAcceptOfferContainerProps
} from '@pages/Movie/components/MovieProfileAcceptOffer/MovieProfileAcceptOfferContainer'
import { String } from '@src/utils'
import { Web3State } from '@state/web3/types'

// ===========================|| ACCEPT OFFER - VIEW ||=========================== //

export type MovieProfileAcceptOfferViewProps = MovieProfileAcceptOfferContainerProps & {
  isLoading: boolean
  handleAcceptOffer: () => void
  buttonSx?: SxProps<Theme>
} & Web3State

const MovieProfileAcceptOfferView: FC<MovieProfileAcceptOfferViewProps> = (props): JSX.Element => {
  const [isOpen, setOpen] = useState(false)
  const onClose = (): void => setOpen(false)
  const onOpen = (): void => setOpen(true)
  const isCompact = props.compact === true ? 'small' : 'large'

  return (
    <>
      <Button
        variant='contained' color='primary' size={isCompact} onClick={() => onOpen()}
        sx={props.buttonSx}
      >
        {
                    props.compact === true
                      ? <Translation target='MOVIE_PROFILE_PRICE_ACCEPT' />
                      : <Translation target='MOVIE_PROFILE_PRICE_ACCEPT_OFFER' />
                }
      </Button>

      <Modal
        isOpen={isOpen}
        onClose={() => onClose()}
      >
        <AcceptOfferContent>
          <Grid container spacing={3}>
            <Grid item xs={12}>
              <header>
                <Typography variant='h3' color='primary'>
                  <Translation target='MOVIE_PROFILE_PRICE_CONFIRMATION' />
                </Typography>
              </header>
            </Grid>
            <Grid item xs={12}>
              <Typography variant='body2' color='text.primary'>
                <Translation target='MOVIE_PROFILE_PRICE_CONFIRMATION_ACCEPT' />
                <strong> {String.minifyHash(props.account)} </strong>
                <Translation target='MOVIE_PROFILE_PRICE_CONFIRMATION_FOR' />
                <strong> {props.price} ETH.</strong>
              </Typography>
            </Grid>
            <Grid item xs={12}>
              {
                                props.result.status > 0
                                  ? <Alert severity='success'>
                                    <Translation target='MOVIE_ACCEPT_OFFER_SUCCESS' />
                                    <strong> Tx: {String.minifyHash(props.result.transactionHash)} </strong>
                                  </Alert>
                                  : props.result.status == 0
                                    ? <Alert severity='error'>
                                      <Translation target='MOVIE_ACCEPT_OFFER_ERROR' />
                                    </Alert>
                                    : <LoadingButton
                                        variant='contained'
                                        color='primary'
                                        size='large'
                                        loading={props.isLoading}
                                        onClick={() => props.handleAcceptOffer()}
                                        fullWidth
                                      >
                                      <Translation target='MOVIE_PROFILE_PRICE_ACCEPT_OFFER' />
                                      </LoadingButton>
                            }

            </Grid>
          </Grid>
        </AcceptOfferContent>
      </Modal>
    </>
  )
}

export default MovieProfileAcceptOfferView

const AcceptOfferContent = styled(Box)<BoxProps>(({ theme }) => ({
  width: '100%',
  maxWidth: '23rem',
  padding: '1.5rem 1.5rem 1.5rem 2rem',
  [theme.breakpoints.down('sm')]: {
    maxWidth: '100%'
  }
}))
