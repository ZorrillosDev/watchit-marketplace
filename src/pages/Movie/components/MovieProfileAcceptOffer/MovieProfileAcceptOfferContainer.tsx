// REACT IMPORTS
import React, { FC, useCallback, useState } from 'react'

// PROJECT IMPORTS
import MovieProfileAcceptOfferView, { MovieProfileAcceptOfferViewProps } from '@pages/Movie/components/MovieProfileAcceptOffer/MovieProfileAcceptOfferView'
import { useEthers } from '@usedapp/core'
import { connect, RootStateOrAny } from 'react-redux'
import { Web3Actions, Web3State } from '@state/web3/types'
import { setApprovalFor } from '@state/web3/actions'
import { selectWeb3Result } from '@state/web3/selector'

// ===========================|| ACCEPT OFFER - CONTAINER ||=========================== //

export type MovieProfileAcceptOfferContainerProps = Omit<MovieProfileAcceptOfferViewProps, 'isLoading' | 'handleAcceptOffer'>

const MovieProfileAcceptOfferContainer: FC<MovieProfileAcceptOfferContainerProps> = (props): JSX.Element => {
  const [isLoading, setIsLoading] = useState(false)
  const { account } = useEthers()
  const { setApprovalFor } = props

  const handleAcceptOffer = useCallback((): void => {
    if (account === undefined) return

    setIsLoading(true)
    setApprovalFor({
      operator: props.account,
      tokenId: props.token,
      approved: props.price.toString()
    })
  }, [account])

  return (
    <MovieProfileAcceptOfferView
      {...{ ...props, isLoading, handleAcceptOffer }
      }
    />
  )
}

const mapDispatchToProps: Partial<Web3Actions> = { setApprovalFor }
const mapStateToProps = (state: RootStateOrAny): Partial<Web3State> => {
  const result = selectWeb3Result(state)
  return { result }
}

export const MovieProfileAcceptOffers = connect(
  mapStateToProps,
  mapDispatchToProps
)(MovieProfileAcceptOfferContainer)
