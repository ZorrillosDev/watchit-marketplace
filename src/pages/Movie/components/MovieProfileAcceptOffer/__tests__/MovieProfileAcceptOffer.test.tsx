import { mount } from 'enzyme'
import React from 'react'
import { Provider } from 'react-redux'
import { store } from '@state/store'
import MovieProfileAcceptOffer from '@pages/Movie/components/MovieProfileAcceptOffer'
<<<<<<< HEAD
=======
import { Web3Actions } from '@state/web3/types'
import { Movie, MovieBid } from '@state/movies/types'
>>>>>>> main
import {
  MovieProfileAcceptOfferContainerProps
} from '@pages/Movie/components/MovieProfileAcceptOffer/MovieProfileAcceptOfferContainer'

/* eslint-disable no-undef */
/* eslint-disable  @typescript-eslint/consistent-type-assertions */

jest.mock('react-router', () => ({
  useParams: () => {
    return { id: '123' }
  }
}))

describe('<MovieProfileAcceptOffer />', () => {
  it('should render', () => {
    const props = {} as MovieProfileAcceptOfferContainerProps
    const component = mount(
      <Provider store={store}>
<<<<<<< HEAD
        <MovieProfileAcceptOffer {...props} account='0x00' />
=======
        <MovieProfileAcceptOffer {...props} />
>>>>>>> main
      </Provider>
    )

    expect(component).toMatchSnapshot()
  })
})
