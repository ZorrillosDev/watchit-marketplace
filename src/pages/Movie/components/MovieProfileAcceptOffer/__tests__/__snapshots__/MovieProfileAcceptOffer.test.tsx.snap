// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieProfileAcceptOffer /> should render 1`] = `ReactWrapper {}`;
